### ISSUE TITLE
Improve/Augment examples

### ISSUE THREAD

**Issue by mwestphal** [f3d-app/f3d#590]
Currently, there are a few examples in https://github.com/f3d-app/f3d/tree/master/examples, but this is very lacking, lets add many useful examples !

**Comment by mwestphal** [comment]
Adding python example should also be considered

**Comment by Ni-g-3l** [comment]
Hey I'm currently playing with python bindings. And I miss a lot some example to understand how it works exactly. 

I'm trying to reproduce actual CPP examples in python (I can put it in a PR if you want). But it's very slow because I miss Python API documentation like class, methods, and attribute (like Read the docs or Furo). May be we could try to add python API doc generated by sphinx to actual F3D documentation (in Bindings section).

And then what kind of example would you add ? (may be I can add it to the PR too in both Python and CPP)

**Comment by mwestphal** [comment]
> May be we could try to add python API doc generated by sphinx to actual F3D documentation (in Bindings section).
 
 Definitely! If you want to work on that you'd be very welcome! You can look at the python tests, its the best we have as a documentation for now.
 
>  And then what kind of example would you add ?

Anything really, copying the C++ examples would be a start

**Comment by mwestphal** [comment]
Python examples have been added, Ill update the issue to have a more coherent plan for examples in general.

**Comment by mwestphal** [comment]
@snoyer @Meakk : please provide more ideas and concrete solutions :)

**Comment by mwestphal** [comment]
Examples are more and more outdated, setting to improve for 3.0 if possible

**Comment by Meakk** [comment]
I think tests within the examples add confusion for the user.  
We should either remove them and test manually the examples before the release, or add all example tests in a separate folder.

**Comment by mwestphal** [comment]
> I think tests within the examples add confusion for the user. We should either remove them and test manually the examples before the release, or add all example tests in a separate folder.

What do you mean add all examples tests in a separate folder ?
Maybe we could add a testing layer in separate files somehow ?

**Comment by Meakk** [comment]
Something like:
```
examples/
├─ cxx/
│  ├─ testing/
│  ├─ example1/
│  ├─ example2/
├─ python/
│  ├─ testing/
│  ├─ example1/
│  ├─ example2/
```

**Comment by mwestphal** [comment]
Sure, that works for me

**Comment by mwestphal** [comment]
Fixed
